src/bucket/BucketSnapshot.cpp

```diff
@@ -101,6 +103,20 @@ BucketSnapshot::loadKeys(std::set<LedgerKey, LedgerEntryIdCmp>& keys,
     auto indexIter = index.begin();
     while (currKeyIt != keys.end() && indexIter != index.end())
     {
+        if (lkMeter)
+        {
+            // If the transactions containing this key have a remaining
+            // quota less than the size of the key, we cannot load the entry.
+            auto keySize = xdr::xdr_size(*currKeyIt);
+            if (!lkMeter->canLoad(*currKeyIt, keySize))
+            {
+                // Consume key size bytes from the quotas of transactions
+                // containing the key (i.e. set their remaining quota to 0).
+                lkMeter->updateReadQuotasForKey(*currKeyIt, keySize);
```

Comment:
Does this mean the key size is only taken into account if we can't load the entry?

Reply:
This is an optimization. At this point we know that the size of the key exceeds the quota, so we do not read the entry. We consume `keySize` (i.e. set the remaining quota to zero), so that next time `canLoad` is called with a key (that only exists in this transaction), it returns false.

Reply:
Ah ok so you use the keySize in `canLoad` because the key is guaranteed to be smaller than the entry, and then you clear the quotas (you're using keySize because you know keySize is enough to clear every quota, but max int would work as well). Maybe clarify this in the comments?

Reply:
I updated the comment to say:

```
                // If the transactions containing this key have a remaining
                // quota less than the size of the key, we cannot load the
                // entry, as xdr_size(key) <= xdr_size(entry). Here we consume
                // keySize bytes from the quotas of transactions containing the
                // key so that they will have zero remaining quota and
                // additional entries belonging to only those same transactions
                // will not be loaded even if they would fit in the remaining
                // quota before this update.                
```
Is that sufficient?